ethereum/p2p/src/main/java/tech/pegasys/pantheon/ethereum/p2p/rlpx/framing/Framer.java

```diff
@@ -256,15 +257,18 @@ private MessageData processFrame(final ByteBuf f, final int frameSize) {
     // Write message data to ByteBuf, decompressing as necessary
     final ByteBuf data;
     if (compressionEnabled) {
-      // Decompress data before writing to ByteBuf
       final byte[] compressedMessageData = Arrays.copyOfRange(frameData, 1, frameData.length - pad);
-      // Check message length
-      Preconditions.checkState(
-          compressor.uncompressedLength(compressedMessageData) < LENGTH_MAX_MESSAGE_FRAME,
-          "Message size in excess of maximum length.");
-      final byte[] decompressedMessageData = compressor.decompress(compressedMessageData);
-      data = NetworkMemoryPool.allocate(decompressedMessageData.length);
-      data.writeBytes(decompressedMessageData);
+      final int uncompressedLength = compressor.uncompressedLength(compressedMessageData);
+      if (uncompressedLength >= LENGTH_MAX_MESSAGE_FRAME) {
+        throw error("Message size %s in excess of maximum length.", uncompressedLength);
+      }
+      try {
+        final byte[] decompressedMessageData = compressor.decompress(compressedMessageData);
+        data = NetworkMemoryPool.allocate(decompressedMessageData.length);
+        data.writeBytes(decompressedMessageData);
+      } catch (final CorruptionException e) {
+        throw new FramingException("Decompression failed", e);
```

Comment:
can this use the "error" call? no one else seems to instantiate one of these explicitly?

Reply:
You can't pass the `CorruptionException` as the cause when using the `error` message (and it can't be overloaded because it's using varargs).